File: MergeToSurvive/src/com/vgdc/merge/entities/abilities/HitboxAbility.java
package com.vgdc.merge.entities.abilities;

import com.badlogic.gdx.math.Vector2;
import com.vgdc.merge.entities.Entity;
import com.vgdc.merge.entities.EntityData;
import com.vgdc.merge.entities.rendering.BlankRenderer;

public class HitBoxAbility extends Ability {
	
	//these 3 variables might be unneccessary, solve later
//	private float duration;
//	private Vector2 dimensions;
//	private int damage;
	//they are
	
	private String hitbox;
	private boolean blank;
	
	public HitBoxAbility(String hitbox)
	{
		this(hitbox, true);
	}
	
	public HitBoxAbility(String hitbox, boolean blank)
	{
		this.hitbox = hitbox;
		this.blank = blank;
	}

	@Override
	public void onUse(Entity entity, boolean retrievable) {
		Entity box = createEntity(entity.getWorld().getHandler().getEntityData(hitbox), entity, retrievable);
		if(blank)
			box.setRenderer(new BlankRenderer());
		box.getMovingBody().setAcceleration(new Vector2(0, 0));
	}
	
	public Entity createEntity(EntityData data, Entity entity, boolean retrievable)
	{
		Entity hitbox = super.createEntity(data, entity, retrievable);
		return hitbox;
	}

}

File: MergeToSurvive-android/assets/assets/scripts/HitBoxAbility.py
from com.vgdc.merge.entities.abilities import Ability
from com.badlogic.gdx.math import Vector2
from com.vgdc.merge.entities import Entity
from com.vgdc.merge.entities import EntityData
from com.vgdc.merge.entities import Projectile
from com.vgdc.merge.entities.rendering import BlankRenderer
	
class HitBoxAbility(Ability):
	
	def __init__(self, hitbox = "", blank = True):
		self.projectile = hitbox
		self.blank = blank

	def onUse(self, entity, retrievable = False):
		box = self.createProjectile(entity.getWorld().getHandler().getEntityData(self.projectile), entity, retrievable)
		if self.blank:
			box.setRenderer(BlankRenderer())
		box.getMovingBody().setAcceleration(Vector2(0, 0))
		
		
